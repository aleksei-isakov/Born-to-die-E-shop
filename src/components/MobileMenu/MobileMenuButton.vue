<template>
  <div class="mobile-menu-button" @click="onClickEmitEvent">
    <div
      v-for="index in elementsNumber"
      :key="index"
      class="mobile-menu-button__element"
    />
  </div>
</template>

<script>
export default {
  name: 'MobileMenuButton',

  data() {
    return {
      elementsNumber: 3
    };
  },

  methods: {
    onClickEmitEvent() {
      this.$emit('click', true);
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.mobile-menu-button {
  display: flex;
  flex-direction: column;
  gap: 6px;
  align-items: center;
  cursor: pointer;

  &__element {
    width: 35px;
    height: 4px;
    background-color: $white;
  }

  @media (min-width: $tablet-size) {
    display: none;
  }
}
</style>
