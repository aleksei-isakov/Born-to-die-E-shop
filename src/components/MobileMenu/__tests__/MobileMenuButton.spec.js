import { shallowMount } from '@vue/test-utils';
import MobileMenuButton from '../MobileMenuButton';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(MobileMenuButton, {
    data() {
      return {
        elementsNumber: 3
      };
    }
  });
  expect(wrapper.is(MobileMenuButton)).toBeTruthy();
});

afterEach(() => {
  wrapper.destroy();
});

describe('MobileMenuButton', () => {
  it('should emit event after user clicked', async () => {
    await wrapper.trigger('click');
    expect(wrapper.emitted().show).toBeTruthy();
  });

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  it('should match a snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
