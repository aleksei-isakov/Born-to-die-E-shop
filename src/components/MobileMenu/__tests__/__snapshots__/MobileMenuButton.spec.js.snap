// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuButton should match a snapshot 1`] = `
<div
  class="mobile-menu-button"
>
  <div
    class="mobile-menu-button__element"
  />
  <div
    class="mobile-menu-button__element"
  />
  <div
    class="mobile-menu-button__element"
  />
</div>
`;
